src/installer/corehost/cli/fxr/shared/CMakeLists.txt

```diff
@@ -17,8 +17,27 @@ set(SOURCES
 
 if(CLR_CMAKE_TARGET_WIN32)
     list(APPEND SOURCES
-        Exports.def)
-endif()
+        hostfxr.def)
+else(CLR_CMAKE_TARGET_WIN32)
+    set(DEF_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/hostfxr_unixexports.src)
+    set(EXPORTS_FILE ${CMAKE_CURRENT_BINARY_DIR}/hostfxr.exports)
+    generate_exports_file(${DEF_SOURCES} ${EXPORTS_FILE})
+
+    if(CLR_CMAKE_HOST_LINUX OR CLR_CMAKE_HOST_FREEBSD OR CLR_CMAKE_HOST_NETBSD)
+        # Add linker exports file option
+        set(EXPORTS_LINKER_OPTION -Wl,--version-script=${EXPORTS_FILE})
```

Comment:
It seems you are missing adding the EXPORTS_LINKER_OPTIONS to the actual linker options like this:
```cmake
set_property(TARGET hostfxr APPEND_STRING PROPERTY LINK_FLAGS ${EXPORTS_LINKER_OPTION})
```